unified-prototype/unified-plugin/plugin-kmp/src/main/java/org/gradle/api/experimental/kotlin/StandaloneKotlinJvmApplicationPlugin.java

```diff
@@ -0,0 +1,39 @@
+package org.gradle.api.experimental.kotlin;
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.experimental.jvm.internal.JvmPluginSupport;
+import org.gradle.api.internal.plugins.software.SoftwareType;
+import org.gradle.api.plugins.ApplicationPlugin;
+import org.gradle.api.plugins.JavaPluginExtension;
+import org.gradle.jvm.toolchain.JavaLanguageVersion;
+import org.jetbrains.kotlin.gradle.dsl.KotlinJvmProjectExtension;
+
+/**
+ * Creates a declarative {@link KotlinJvmApplication} DSL model, applies the official Kotlin and application plugin,
+ * and links the declarative model to the official plugin.
+ */
+abstract public class StandaloneKotlinJvmApplicationPlugin implements Plugin<Project> {
+    @SoftwareType(name = "kotlinJvmApplication", modelPublicType = KotlinJvmApplication.class)
+    abstract public KotlinJvmApplication getApplication();
+
+    @Override
+    public void apply(Project project) {
+        KotlinJvmApplication dslModel = getApplication();
+
+        project.getPlugins().apply(ApplicationPlugin.class);
+        project.getPlugins().apply("org.jetbrains.kotlin.jvm");
+
+        linkDslModelToPluginLazy(project, dslModel);
+    }
+
+    private void linkDslModelToPluginLazy(Project project, KotlinJvmApplication dslModel) {
```

Comment:
💭 I think we can drop the "Lazy" suffix from these methods, they don't seem to be doing anything lazily ATM.

Reply:
I'll do this in the next PR.